source/src/behaviour/cs_BehaviourConflictResolution.cpp

```diff
@@ -1,5 +1,49 @@
 #include <behaviour/cs_BehaviourConflictResolution.h>
 
+/*
+ * returns a 'relevance', lower is less relevant.
+ *  - VacuouslyTrue is least relevant.
+ *  - 'in Sphere' is less relevant than 'In room'.
+ *  - 'presence' is less relevant than 'absence'.
+ */
+auto getRelevance(PresencePredicate::Condition condition) {
+    switch(condition) {
+        case PresencePredicate::Condition::VacuouslyTrue: return 0;
+        case PresencePredicate::Condition::AnyoneInSphere: return 1;
+        case PresencePredicate::Condition::NooneInSphere: return 2;
+        case PresencePredicate::Condition::AnyoneInSelectedRooms: return 3;
+        case PresencePredicate::Condition::NooneInSelectedRooms: return 4;
+        default: return -1; // worst relevance: unknown enum value
+    }
+};
+
+bool isRoomBased(PresencePredicate predicate) {
+    return predicate._condition == PresencePredicate::Condition::AnyoneInSelectedRooms
+        || predicate._condition == PresencePredicate::Condition::NooneInSelectedRooms;
+}
+
+int numberOfRooms(PresencePredicate predicate) {
+    uint64_t bitmask = predicate._presence.getBitmask();
+    int count = 0;
+    for (auto i{0}; i<8; i++){
```

Comment:
I wouldn't have guessed `8` here in the for-loop because bitmask is `64`-bits. Perhaps indicate that this is done on purpose or get the magic number `8` from somewhere.

Reply:
Yup. Bart found the same bug.